scripts/ingest_regress_test_result-new-format.py

```diff
@@ -0,0 +1,198 @@
+#! /usr/bin/env python3
+
+import argparse
+import json
+import logging
+import os
+import re
+import sys
+from collections import namedtuple
+from contextlib import contextmanager
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Tuple
+
+import backoff
+import psycopg2
+from psycopg2.extras import execute_values
+
+CREATE_TABLE = """
+CREATE TABLE IF NOT EXISTS results (
+    id           BIGSERIAL PRIMARY KEY,
+    parent_suite TEXT NOT NULL,
+    suite        TEXT NOT NULL,
+    name         TEXT NOT NULL,
+    status       TEXT NOT NULL,
+    start_ts     TIMESTAMP NOT NULL,
+    stop_ts      TIMESTAMP NOT NULL,
+    duration_s   INT NOT NULL,
+    flaky        BOOLEAN NOT NULL,
+    build_type   TEXT NOT NULL,
+    pg_version   INT NOT NULL,
+    run_id       BIGINT NOT NULL,
+    run_attempt  INT NOT NULL,
+    reference    TEXT NOT NULL,
+    revision     CHAR(40) NOT NULL,
+    raw          JSONB COMPRESSION lz4 NOT NULL,
+    UNIQUE (parent_suite, suite, name, build_type, pg_version, start_ts, stop_ts, run_id)
+);
+"""
+
+Row = namedtuple(
```

Comment:
Lets use NamedTuple from `typing` or `dataclass` module.

Also it would be great to have short doc comments for fields to clarify their meaning.

You also can use the class to define methods, e g something like:

```python
@classmethod
def from_json(cls, ..)
```

Reply:
I have changed to `dataclass`. Thanks for the suggestion!